cmd/utils/nodecmd/chaincmd.go

```diff
@@ -74,73 +75,51 @@ func initGenesis(ctx *cli.Context) error {
 	genesisPath := ctx.Args().First()
 	if len(genesisPath) == 0 {
-		log.Fatalf("Must supply path to genesis JSON file")
+		logger.Crit("Must supply path to genesis JSON file")
 	}
 	file, err := os.Open(genesisPath)
 	if err != nil {
-		log.Fatalf("Failed to read genesis file: %v", err)
+		logger.Crit("Failed to read genesis file", "err", err)
 	}
 	defer file.Close()
 
 	genesis := new(blockchain.Genesis)
 	if err := json.NewDecoder(file).Decode(genesis); err != nil {
-		log.Fatalf("invalid genesis file: %v", err)
+		logger.Crit("invalid genesis file", "err", err)
 		return err
 	}
 
-	genesis = checkGenesisAndFillDefaultIfNeeded(genesis)
-
-	if genesis.Config.Istanbul != nil {
-		if err := governance.CheckGenesisValues(genesis.Config); err != nil {
-			logger.Crit("Error in genesis json values", "err", err)
-		}
-
-		// Check if governingNode is properly set
-		if strings.ToLower(genesis.Config.Governance.GovernanceMode) == "single" {
-			if istanbulExtra, err := decodeExtra(genesis.ExtraData); err != nil {
-				logger.Crit("Extra data couldn't be decoded. Please check if your extra data is correct", "err", err)
-			} else {
-				var found bool
-				for _, v := range istanbulExtra.Validators {
-					if v == genesis.Config.Governance.GoverningNode {
-						found = true
-						break
-					}
-				}
-				if !found {
-					logger.Crit("GoverningNode is not in the validator list. Please check if your governingNode address is correct")
-				}
-			}
-		}
-	}
-
-	if genesis.Config.Governance.Reward.StakingUpdateInterval != 0 {
-		params.SetStakingUpdateInterval(genesis.Config.Governance.Reward.StakingUpdateInterval)
-	} else {
-		genesis.Config.Governance.Reward.StakingUpdateInterval = params.StakingUpdateInterval()
+	if genesis.Config == nil {
+		logger.Crit("invalid genesis config")
 	}
 
-	if genesis.Config.Governance.Reward.ProposerUpdateInterval != 0 {
-		params.SetProposerUpdateInterval(genesis.Config.Governance.Reward.ProposerUpdateInterval)
-	} else {
-		genesis.Config.Governance.Reward.ProposerUpdateInterval = params.ProposerUpdateInterval()
+	// Update undefined config with default values and validate config
+	genesis.Config.Enhance()
+	if err := ValidateGenesisConfig(genesis); err != nil {
+		logger.Crit("invalid genesis", "err", err)
 	}
 
-	data := getGovernanceItemsFromGenesis(genesis)
-	gbytes, err := json.Marshal(data.Items())
+	// Set genesis.Governance and reward intervals
+	govSet := getGovernanceItemSetFromGenesis(genesis)
+	govItemBytes, err := json.Marshal(govSet.Items())
 	if err != nil {
 		logger.Crit("Failed to json marshaling governance data", "err", err)
 	}
-	if genesis.Governance, err = rlp.EncodeToBytes(gbytes); err != nil {
+	if genesis.Governance, err = rlp.EncodeToBytes(govItemBytes); err != nil {
 		logger.Crit("Failed to encode initial settings. Check your genesis.json", "err", err)
 	}
+
+	params.SetStakingUpdateInterval(genesis.Config.Governance.Reward.StakingUpdateInterval)
```

Comment:
Is it ok not to check if it is 0 or not?

Reply:
If `genesis.Config.Governance.Reward.StakingUpdateInterval` is 0, it will be updated to the default value at `genesis.Config.Enhance()` (line 96)

Reply:
I also added validation logic for the interval values in `ValidateGenesisConfig` to make sure.